classes/Language.php

```diff
@@ -882,6 +882,8 @@ public static function checkAndAddLanguage($iso_code, $lang_pack = false, $only_
             }
         }
 
+        self::loadLanguages();
```

Comment:
If you add a new language through checkAndAddLanguage, it will not update the local cache. Hence If you're using getLanguages() just after adding a language with checkAndAddLanguage(), the new language will not be returned.